packages/valory/skills/abstract_round_abci/behaviour_utils.py

```diff
@@ -1112,6 +1130,43 @@ def _get_status(self) -> Generator[None, None, HttpMessage]:
         result = yield from self._do_request(request_message, http_dialogue)
         return result
 
+    def _get_netinfo(
+        self, timeout: Optional[float] = None
+    ) -> Generator[None, None, HttpMessage]:
+        """Makes a GET request to it's tendermint node's /net_info endpoint."""
+        request_message, http_dialogue = self._build_http_request_message(
+            method="GET", url=f"{self.context.params.tendermint_url}/net_info"
+        )
+        result = yield from self._do_request(request_message, http_dialogue, timeout)
+        return result
+
+    def num_active_peers(
+        self, timeout: Optional[float] = None
+    ) -> Generator[None, None, Optional[int]]:
+        """Returns the number of active peers in the network."""
+        try:
+            http_response = yield from self._get_netinfo(timeout)
+            http_ok = 200
+            if http_response.status_code != http_ok:
+                # a bad response was received, we cannot retrieve the number of active peers
+                self.context.logger.warning(
+                    f"/net_info responded with status {http_response.status_code}"
+                )
+                return None
+
+            res_body = json.loads(http_response.body)
+            num_peers = int(res_body["result"]["n_peers"])
```

Comment:
Do you think that just in case we should do:

```suggestion
            num_peers = int(res_body.get("result", {}).get("n_peers", None))
            if num_peers is None:
                return None
```

or would that be redundant?

Reply:
I'd say your suggestion makes it better, I will add it.